bin/live/pycbc_live_collate_triggers

```diff
@@ -0,0 +1,333 @@
+        end_time = start_time + duration
+
+        if file_count % 100 == 0:
+            logging.info(f" Files appended: {file_count}/{len(trigger_files)}")
+
+        with h5py.File(source_file, 'r') as source:
+            for ifo in args.ifos:
+                triggers = source[ifo]
+                if ('approximant' not in triggers) or (len(triggers['approximant']) == 0):
+                        continue
+
+                for name, dataset in triggers.items():
+
+                    if name in destination[ifo]:
+                        if triggers[name].shape[0] == 0:
+                            continue
+                        # Append new data to existing dataset in destination
+                        destination[ifo][name].resize((destination[ifo][name].shape[0] + triggers[name].shape[0]), axis=0)
+                        destination[ifo][name][-triggers[name].shape[0]:] = triggers[name]
+                    else:
+                        destination[ifo].create_dataset(name, data=dataset, chunks=True, maxshape=(None,))
+
+                for attr_name, attr_value in source.attrs.items():
+                    destination.attrs[attr_name] = attr_value
+
+                # Create or get the '/search' subgroup within 'H1'/'L1'/'V1'
+                if 'search' not in destination[ifo]:
+                    destination[ifo].create_group('search')
+
+                # Create or append 'start_time' and 'end_time' datasets within the '/search' group
+                if 'start_time' in destination[ifo]['search']:
+                    destination[ifo]['search']['start_time'].resize((destination[ifo]['search']['start_time'].shape[0] + 1), axis=0)
+                    destination[ifo]['search']['start_time'][-1] = start_time
+                else:
+                    destination[ifo]['search'].create_dataset('start_time', chunks=True, data=numpy.array([start_time]), maxshape=(None,))
+
+                if 'end_time' in destination[ifo]['search']:
+                    destination[ifo]['search']['end_time'].resize((destination[ifo]['search']['end_time'].shape[0] + 1), axis=0)
+                    destination[ifo]['search']['end_time'][-1] = end_time
+                else:
+                    destination[ifo]['search'].create_dataset('end_time', chunks=True, data=numpy.array([end_time]), maxshape=(None,))
+
+# Add all the template boundaries
+with h5py.File(output_file, 'a') as output:
+    for ifo in args.ifos:
+        logging.info(f" Adding template boundaries for {ifo}")
+        # Grab the ifo you want
+        triggers = output[ifo]
+        try:
+            template_ids = triggers['template_id'][:]
+        except:
+            logging.info(f"  No triggers for {ifo}, skipping")
+            continue
+
+        sorted_indices = numpy.argsort(template_ids)
+        sorted_template_ids = template_ids[sorted_indices]
+        unique_template_ids, template_id_counts = numpy.unique(sorted_template_ids, return_counts=True)
```

Comment:
This line here doesn't include the `0` template_id_counts. I think you'd want to add that in.

Reply:
That's a separate issue (though it is fixed a couple of lines down) - I think the problem is that the region references are not being created further down, so when the fit_by_template script tries to call them, it fails